### PR TITLE
Truncate the statefulset pvc name to be max 63 chars

### PR DESCRIPTION
**Pull request by tjhiggins**
Changes proposed in this PR:
- https://github.com/hashicorp/consul-k8s/issues/798#issuecomment-947907238

How I expect reviewers to test this PR:
- Run the server-statefulset.bats unit tests

### CODE REVIEW

**Comment by lkysow** [comment]
Looks great! Can you write a Helm test for it?

**Comment by tjhiggins** [comment]
@lkysow done

File: charts/consul/test/unit/server-statefulset.bats [hunk 1]
```diff
@@ -138,6 +138,28 @@ load _helpers
...
+      -s templates/server-statefulset.yaml  \
+      . | tee /dev/stderr |
+      yq -r '.spec.volumeClaimTemplates[0].metadata.name' | tee /dev/stderr)
+  [ "${actual}" = "data-default" ]
+}
+
+@test "server/StatefulSet: truncation for namespace > 58 chars" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -n really-really-really-really-really-really-really-long-namespace \
```
Comment (lkysow) [inline charts/consul/test/unit/server-statefulset.bats:156]: 😄

**Review by lkysow — APPROVED** [review]
Looks great!! Thanks so much.

**Pull request merged**